prism-agent/service/server/src/main/scala/io/iohk/atala/LogUtils.scala

```diff
@@ -0,0 +1,23 @@
+package io.iohk.atala
+
+import zio.*
+import io.iohk.atala.api.http.RequestContext
+import sttp.model.Header
+
+object LogUtils {
+  inline val headerName = "X-Request-ID"
+
+  extension [R, E, A](job: ZIO[R, E, A])
+    inline def logTrace(ctx: RequestContext)(implicit trace: Trace): ZIO[R, E, A] = {
+      // TODO We can add a timer here
+      // TODO We can also send metric (couter and/or timer) to another system from here
```

Comment:
@davidpoltorak-io What are your feelings about this?

Reply:
When using prometheus and grafana and loki - we can already count the number of lines at each log level as long as they use a standard format and loki / grafana is configured to extract it.

We could also add an explicit metric - having a `trace counter` or `trace gauge` which can show the number of times this is called and the average duration would be useful for anyone not running the logging stack